### ISSUE TITLE
[BUG] avoid-redundant-async false positive

### ISSUE THREAD

**Issue by praxa-enzo** [dart-code-checker/dart-code-metrics#1223]
### Environment and configuration

DCM version: 5.7.2
Dart SDK version: 2.19.6

<details>
<summary>Configuration</summary>

```yaml
name: bug
description: A sample command-line application.
version: 1.0.0

environment:
  sdk: '>=2.19.6 <3.0.0'

dev_dependencies:
  dart_code_metrics: 5.7.2
  lints: ^2.0.0
  test: ^1.21.0

```
</details>

### What did you do?

```dart
Future<void> foo() async {
  await for (final value in Stream.fromIterable([0, 1, 2])) {
    print(value);
  }
}
```

### What did you expect to happen?

No warnings.

### What actually happened?

Linter warning: 'async' keyword is redundant, consider removing it.

### Participation

- [ ] I am willing to submit a pull request for this issue.
